### ISSUE TITLE
Feedback and Suggestions

### ISSUE THREAD

**Issue by GrechHerald** [zaeonNineZero/NineZero-Gun-Expansion#20]
General feedback for CGM AVE and NZGE:

- I find the common reload sounds a bit limiting for resource pack making, I have mentioned this on discord in the past but I want to mention it again just in case; I understand however that it was done to optimize filesize
- I think the CGM Shotgun remodel is a bit too similar to the Pump Shotgun; it'd be nicer if there was a more clear visual distinction between the two

Now for the NZGE suggestions:

- more grenade types would be really nice to have, not sure what's planned to come but some example I'd love to see are smoke grenades, incendiary grenades and molotovs, though incendiary grenades might just be redundant
- if possible, underbarrel grenade launchers? with a different visual style depending on what type of grenade they fire (HE launcher, smoke launcher, even if a bit strange flashbang launcher, etc.)
- Compensator Attachment, cheaper to make than a muzzle break but only reduces vertical recoil (if possible) (if not, reduces recoil significantly less) (bit more on this, since there's the idea for a heavy SMG modeled after the Thompson, I think a Cutts Compensator would be most fitting)
- this one is more for nostalgia sake but what if the Infantry Rifle got a Ghillie Mesh when the Ballistic Scope and Suppressor are attached? kind of a reference to CoD4 MW more or less
- laser sight attachment for the underbarrel slot, just reduces spread a tiny bit (similar to what CGM Refined had)
- perhaps a classic flamethrower weapon so the player doesn't have to get the fire starter enchantment?
- also on the uncertain part but what about a PDW type weapon? I know you did make the MP7(A2) for the 1.4 update which could fill a similar role but I was originally thinking of more of a P90 kind of deal that fires fast but does more damage per shot
- an in-between the lightweight and tactical stocks, a folding stock, even if it may not be able to actually fold; visual inspiration for it could be taken from the FN FNC, AKS-74(U), or the SIG SG 550 (or anything else you may have in mind)
- a Tactical Grip that combines both the laser sight and the vertical grip but is more expensive to make and reduces ADS speed due to added weight (see attached image below for this one specifically; I believe it might be called the "Osprey Tactical Grip" irl but I'm not really sure (image is from Insurgency: Sandstorm, attached on the bottom of an HS Produkt VHS-2))
![image](https://github.com/zaeonNineZero/NineZero-Gun-Expansion/assets/137415131/7a312823-6525-4df9-b69c-99f0a6d1888a)

These are all of my suggestions for the future of NZGE, certainly take your time with these because it is relatively speaking a lot of things
thank you for making this awesome addon

**Comment by zaeonNineZero** [comment]
Phew, that's a lot of stuff! I'll do my best to go through it all.

Some early answers:

- CGM on its own has only one reload sound, which is very limiting. I added my extra reload sounds based on a simple evaluation of a weapon's role and ammo type - so the pistols and SMGs share the default "pistol" reload sound, and unique reload sounds are added for the shotgun, (semi-)automatic rifle, and sniper rifle weapon roles. This wouldn't be too hard to expand this for the purposes of resource pack making (from a few new sound types for the different weapon roles, to each weapon having a unique sound event that pulls from common sound files in sounds.json). I could also take a hybrid approach between the two options, so let me know what would work best from your POV.
- Noted. Do you have any suggestions on what could be done to differentiate it further from the Pump Shotgun? (The CGM-A/VE Shotgun model was partially based on the M1014, maybe an element of that shotgun or similar ones could help make it more visually distinct?)

1. Definitely possible. I'll keep peeking around in the CGM source code to see what I can do - the toughest part is probably going to be handling the "explosion" events, rather than adding new items and entities.
2. Don't think this is possible without a **very** substantial rework of CGM's code. While I do have plans for a mixin-driven addon to add new features to CGM, secondary-fire attachments would require a massive rework of the mod's code to support it.
3. That's something that could be added in the near future (post v1.4.0). It would probably come with stat tweaks to the muzzle brake to ensure their functions don't overlap too much.
4. I'm admittedly not a fan of the ghillie mesh aesthetic for one, but more importantly, I think that breaks too far away from the model style of CGM and NZGE.
5. Possibly, wouldn't be too hard to make a basic model and give it stats. I might wait on it though to see if I can/can't do anything fancy with it.
6. Flamethrower would be a very nice addition, but I'd want to do something fancy with it if I made one. (requires a liquid like lava as "fuel"? or fully reload it with a single "canister" item?) Might warrant being an independent addon at that point, but I'll keep it mind nonetheless.
7. Could do that, and since the P90 uses FN 5.7×28mm ammunition maybe it could be a counterpart to the Rapid SMG and Micro SMG, firing Medium Bullets rather than Basic Bullets. Would be a great way to differentiate it from the other two bullet hoses.
8. That's funny, I've actually been thinking about adding a folding-style stock as an attachment for the past several weeks now. I can definitely fit that in as post v1.4.0 content.
9. See above - if I add laser sights then this could be something I do. Unless the laser sight becomes a "tactical slot" attachment in which case a laser grip would be redundant.

Thanks again for the feedback! I'll do a lot of thinking about these extra ideas you've floated.

**Comment by GrechHerald** [comment]
for the CGM shotgun, I think making it more resemble the Benelli M4 Super 90 (european name for the M1014) some more would help it out actually, maybe with a bit of inspiration taken from the Benelli M3 Super 90 (which is a pump action and semi-auto shotgun hybrid)

can you describe a bit more about this hybrid system? could probably work fine, I've worked with "sounds.json" files before

I see on the grenade launchers, bit unfortunate but I do understand it would basically need forking

fair on the Ghillie Mesh, I might make a very basic one myself in the future for personal use

on the laser sight, I was thinking of it as a choice thing between a grip and itself with the laser grip coming in later as a potential solution (pretty much based on what's already existing in baseline CGM) but if you can manage to make a tactical attachment that'd be nice too

flamethrower could use a single canister item I think that you can craft it with blaze powder and stuff

precisely what I was thinking on the PDW, something that like does 4.8-5.3 damage per shot (if considering an assault rifle with actual balance changes does about 5.6 damage) chambered in medium bullets, awesome to see we both basically thought of the same thing more or less

you're welcome, love your mod

**Comment by zaeonNineZero** [comment]
1. I'll also add some of the changes we tossed around in the CGM Discord too.
2. Sounds that will be played by mods (and not just by commands) need to have a sound event registered, not just defined in the sounds.json. You can see [here](https://github.com/zaeonNineZero/NineZero-Gun-Expansion/blob/latest/src/main/java/zaeonninezero/nzgmaddon/init/initSounds.java) how NZGE does so. Basically, I can add reload sound events for every gun in NZGE, and retroactively add one for use with CGM's Machine Pistol. Then in sounds.json I can set the sound events to use common reload sound files, which can then be remapped by soundpack creators to other sound files.

5. The route I take depends on whether I implement an additional attachment category or not (not directly through NZGE at least). I have not come to a verdict on that, but implementing a laser sight as an underbarrel attachment does not necessarily preclude it from being reworked into a tactical attachment down the road.
6. That is one option. I'd want the canister to fully reload the flamethrower with just one item, so I'd need to do some mixin work on CGM's code to make that work. That might get resolved with another addon mod though.
7. I was thinking lower damage (>=3.5 damage) with a maxed-out rate of fire like the Micro/Rapid SMGs. It would then fill the same rapid-fire CQC role with a more expensive ammo type, and in return have a better DPS. (Maybe at the cost of accuracy or recoil, nothing is set in stone just yet)

**Comment by GrechHerald** [comment]
I see I see, thanks again

**Comment by zaeonNineZero** [comment]
A few updates related to these suggestions, taking my CGM Expanded fork into consideration:

- The new animation system and new reload sound mechanics in CGM Expanded have allowed me to add several new sounds that go with these animations. The common reload sounds will be used when base CGM is used, while the new sounds will replace them when CGM Expanded is loaded.
- I've been doing some thinking about new attachment slot types (gadget/tactical, magazine), which would allow adding a laser sight as a possible attachment. I still plan on trying out some fancy ideas with it to make it stand out.
- Flamethrower that loads from a single canister is 100% possible thanks to CGM Expanded. The projectile override system will also allow for a custom flame projectile to be used. Definitely going to be a standalone addon from NZGE though, since it'll depend significantly on CGM Expanded.
- A Laser Grip could be a possibility if additional gadget attachments besides the laser sight are added.

**Comment by zaeonNineZero** [comment]
(Did not mean to close this issue - reopened it since it's still relevant)

**Comment by GrechHerald** [comment]
awesome

**Comment by Galaxy7172** [comment]
Heya, I've been using your mods for a while, I saw some of the feedback and suggestions; I've decided to come up with some of my own:

Skins for the Rapid SMG(NZGE):

1. SR-2 Veresk [https://en.wikipedia.org/wiki/SR-2_Veresk](url)
2. PP-2000 [https://en.wikipedia.org/wiki/PP-2000](url) _(Best contender)_
3. PP-16 Bizon [https://en.wikipedia.org/wiki/PP-19_Bizon](url) _(Best contender)_

Skins/Replacement Model for the Pistol(CGM/CGMe):

1. SR-1 Vector _(could add a dovetail mount for the pistol when adding sights)_ [https://en.wikipedia.org/wiki/SR-1_Vektor#SR-1M](url)

Skin/Replacement Model for the Machine Pistol(CGM/CGMe):

1. PP-91 Keder [https://en.wikipedia.org/wiki/PP-91_Kedr](url)
